app/presenters/tree_builder_alert_profile_obj.rb

```diff
@@ -0,0 +1,59 @@
+class TreeBuilderAlertProfileObj < TreeBuilder
+  def initialize(name, type, sandbox, build = true, assign_to: nil, cat: nil, objects: nil)
+    @assign_to = assign_to
+    @cat = cat
+    @objects = objects
+    @cat_tree = true if @assign_to.ends_with?("-tags")
+    super(name, type, sandbox, build)
+  end
+
+  def override(node, object, _pid, _options)
+    identifier = (object.name.presence || object.description)
+    identifier += "-" + object.hostname if object.kind_of?(MiddlewareServer)
```

Comment:
sounds like this should be `identifier = "#{identifier} - #{object.hostname}" if object.kind_of?(MiddlewareServer)`

But it's more broken than that :)... if you get an instance of `MiddlwareServer` somewhere, the tree will try to create a `TreeNode::MiddlewareServer` instance for `node` .. and fail because there is no `TreeNode::MiddlewareServer`. 
So you'll need to create one, if this tree needs to support `MiddlewareServer`. And once you've created it, the line can live there :).